packages/elements/src/components/RequestSamples/RequestSamples.tsx

```diff
@@ -36,24 +36,35 @@ export const RequestSamples = React.memo<RequestSamplesProps>(({ request }) => {
 
   const requestSample = convertRequestToSample(httpSnippetLanguage, httpSnippetLibrary, request);
 
-  const handleSelectClick: SelectProps['onChange'] = (value: string | number) => {
-    const [language, library] = String(value).split(' / ');
-    setSelectedLanguage(language);
-    setSelectedLibrary(library || '');
-  };
-
   return (
     <Panel rounded isCollapsible={false}>
       <Panel.Titlebar rightComponent={<CopyButton size="sm" copyValue={requestSample || ''} />}>
         <Box ml={-2}>
-          <Select
-            aria-label="Request Sample Language"
-            onChange={handleSelectClick}
-            options={selectOptions}
-            value={selectedLibrary ? `${selectedLanguage} / ${selectedLibrary}` : selectedLanguage}
-            triggerTextPrefix="Request Sample: "
-            size="sm"
-          />
+          <Menu
+            label="Request Sample Language"
+            trigger={
+              <Button size="sm" iconRight="caret-down">
+                Request Sample: {selectedLanguage} / {selectedLibrary}
+              </Button>
+            }
+          >
+            {Object.entries(requestSampleConfigs).map(([language, config]) => (
+              <MenuItem key={language} indent text={language} checked={selectedLanguage === language}>
+                {Object.keys(config.libraries ?? {}).map(library => (
```

Comment:
```suggestion
                {config.libraries && Object.keys(config.libraries).map(library => (
```

That would prevent rendering empty array